src/stream/stream_backend.rs

```diff
@@ -316,24 +322,29 @@ fn create_rtsp_stream(
 
     let video_format = match video_source {
         VideoSourceType::Local(local_device) => {
-            if VideoEncodeType::H264 == encode {
-                format!(
-                    concat!(
-                        "v4l2src device={device}",
-                        " ! video/x-h264,width={width},height={height},framerate={interval_denominator}/{interval_numerator},type=video",
-                    ),
-                    device = &local_device.device_path,
-                    width = configuration.width,
-                    height = configuration.height,
-                    interval_denominator = configuration.frame_interval.denominator,
-                    interval_numerator = configuration.frame_interval.numerator,
-                )
-            } else {
-                return Err(SimpleError::new(format!(
-                    "Unsupported encode for RTSP endpoint: {:?}",
-                    encode
-                )));
-            }
+            let capability = match &encode {
+                VideoEncodeType::H264 => "video/x-h264",
+                VideoEncodeType::YUYV => "video/x-raw,format=YUY2",
+                VideoEncodeType::MJPG => "image/jpeg",
```

Comment:
We should put that on a function that transform type to gstreamer type, this will remove the redundant text.